tests/storage/migration.go

```diff
@@ -506,6 +506,27 @@ var _ = SIGDescribe("[Serial]Volumes update with migration", Serial, func() {
 			Expect(err).ToNot(HaveOccurred())
 			checkVolumeMigrationOnVM(vm, volName, dv.Name, destPVC)
 		})
+
+		It("should cancel the migration should clear the volume migration information on the VM", func() {
+			volName := "volume"
+			dv := createDV()
+			vm := createVMWithDV(dv, volName)
+			createUnschedulablePVC(destPVC, ns, size)
+			By("Update volumes")
+			updateVMWithPVC(vm, volName, destPVC)
+			waitMigrationToExist(vm.Name, ns)
+			waitVMIToHaveVolumeChangeCond(vm.Name, ns)
+			By("Cancel the volume migration")
+			updateVMWithPVC(vm, volName, dv.Name)
+			Eventually(func() *virtv1.VolumeMigrationState {
+				vm, err := virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				if vm.Status.VolumeUpdateState == nil {
+					return nil
+				}
+				return vm.Status.VolumeUpdateState.VolumeMigrationState
+			}).WithTimeout(120 * time.Second).WithPolling(time.Second).Should(BeNil())
```

Comment:
Shouldn't we check that the `vm.Status.VolumeUpdateState.VolumeMigrationState` was filled out before checking if it was gone?
Since `vm.Status.VolumeUpdateState` is nil at the beginning the eventually should succeed just because it was never populated, or am I missing something?

Reply:
/hold
I think so too